packages/rehype-shikiji/src/core.ts

```diff
@@ -108,8 +117,8 @@ const rehypeShikijiFromHighlighter: Plugin<[HighlighterGeneric<any, any>, Rehype
         return
       }
 
-      const attrs = (head.data as any)?.meta
-      const meta = parseMetaString?.(attrs, node, tree) || {}
+      const attrs = head.data?.meta ?? head.properties.metastring
+      const meta = (attrs && parseMetaString?.(attrs, node, tree)) || {}
```

Comment:
I think it's better to always pass it to the function even the metastring is not presented, so users could still have a chance to provide their meta.